src/DependencyFactory.php

```diff
@@ -145,7 +145,7 @@ public function getConfiguration(): Configuration
     {
         if ($this->configuration === null) {
             $this->configuration = $this->configurationLoader->getConfiguration();
-            $this->freeze();
+            $this->frozen = true;
```

Comment:
Why these changes where the `frozen()` method gets replaced with the direct set to true?

Reply:
PHPStan treats the method calls as impure and complains that `$this->configuration` might be `null` when we return it.

Reply:
I was playing around with annotations and other configurations, but it seems like PHPStan can't be convinced that `$this->freeze()` does nothing evil with other property states.

Reply:
Yes, and this is fine imho. `freeze()` is impure, so PHPStan's behavior is correct.